torchaudio/models/wav2letter.py

```diff
@@ -0,0 +1,71 @@
+        version (str, optional): Wav2Letter can use as input: ``waveform``, ```power_spectrum``
+         or ```mfcc``. (Default: ``waveform``)
+        n_input_features (int, optional): If is used ``power_spectrum`` or ```mfcc`` must be
+        specified the number of features that were extracted. (Default: ``None``)
+    """
+
+    def __init__(self, num_classes=40, version="waveform", n_input_features=None):
+        super(Wav2Letter, self).__init__()
+        n_input_features = 250 if not n_input_features else n_input_features
+
+        acoustic_model = nn.Sequential(
+            nn.Conv1d(in_channels=n_input_features, out_channels=250, kernel_size=48, stride=2, padding=23),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=250, out_channels=250, kernel_size=7, stride=1, padding=3),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=250, out_channels=250, kernel_size=7, stride=1, padding=3),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=250, out_channels=250, kernel_size=7, stride=1, padding=3),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=250, out_channels=250, kernel_size=7, stride=1, padding=3),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=250, out_channels=250, kernel_size=7, stride=1, padding=3),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=250, out_channels=250, kernel_size=7, stride=1, padding=3),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=250, out_channels=250, kernel_size=7, stride=1, padding=3),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=250, out_channels=2000, kernel_size=32, stride=1, padding=16),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=2000, out_channels=2000, kernel_size=1, stride=1, padding=0),
+            nn.ReLU(inplace=True),
+            nn.Conv1d(in_channels=2000, out_channels=num_classes, kernel_size=1, stride=1, padding=0),
+            nn.ReLU(inplace=True)
+        )
+
+        if version == "waveform":
+            waveform_model = nn.Sequential(
+                nn.Conv1d(in_channels=1, out_channels=250, kernel_size=250, stride=160, padding=45),
+                nn.ReLU(inplace=True)
+            )
+            self.acoustic_model = nn.Sequential(waveform_model, acoustic_model)
+
+        if version in ["power_spectrum", "mfcc"]:
+            self.acoustic_model = acoustic_model
+
+    def forward(self, x):
+        # type: (Tensor) -> Tensor
+        r"""
+        Args:
+            x (torch.Tensor): Tensor of dimension (batch_size, n_features, input_length).
+
+        Returns:
+            torch.Tensor: Predictor tensor of dimension (input_length, batch_size, number_of_classes).
+        """
+
+        x = self.acoustic_model(x)
+        x = nn.functional.log_softmax(x, dim=1)
+        x = x.permute(2, 0, 1)
```

Comment:
That's useful for CTC/ASG, but we should be careful around standardizing for this here. I'm tempted to leave the output shape by default for any models as `batch, class, length`, and let the user do what they want. DataParallel in particular breaks batches on the first dimension by default. This may not play as nicely with LSTM/etc though. Thoughts?

Reply:
Removed to allow more control for the user